packages/flutter/lib/src/widgets/actions.dart

```diff
@@ -355,6 +361,183 @@ class Actions extends InheritedWidget {
   }
 }
 
+/// A widget that combines the functionality of [Actions], [Shortcuts],
+/// [MouseRegion] and a [Focus] widget to create a detector that defines actions
+/// and key bindings, and will notify that the focus or hover highlights should
+/// be shown or not.
+///
+/// This widget can be used to give a control the required detection modes for
+/// focus and hover handling.
+///
+/// This widget doesn't have any visual representation, it is just a detector that
+/// provides focus and hover capabilities.
+///
+/// It hosts its own [FocusNode] or uses [focusNode], if given.
+class FocusableActionDetector extends StatefulWidget {
```

Comment:
I like the concept, but don't love the name `FocusableActionDetector`. However, as all the suggestions I could come up with are worse, perhaps I should just keep this to myself :).

Reply:
Yeah, I don't love the name either.  It's the best I could come up with, though: it has to do with focus, actions, shortcuts, and hover, and it doesn't have a visual representation. `FocusableShortcutActionsAndHoverDetector` seemed like a mouthful. :-)